pkg/kritis/violation/strategy.go

```diff
@@ -16,29 +16,65 @@ limitations under the License.
 package violation
 
 import (
-	"github.com/grafeas/kritis/pkg/kritis/metadata"
+	"github.com/grafeas/kritis/pkg/kritis/constants"
+	"github.com/grafeas/kritis/pkg/kritis/crd/securitypolicy"
+	"github.com/grafeas/kritis/pkg/kritis/pods"
 	"github.com/sirupsen/logrus"
 	"k8s.io/api/core/v1"
 )
 
 type Strategy interface {
-	HandleViolation(image string, pod *v1.Pod, violations []metadata.Vulnerability) error
+	HandleViolation(image string, pod *v1.Pod, violations []securitypolicy.SecurityPolicyViolation) error
 }
 
 type LoggingStrategy struct {
 }
 
-func (l *LoggingStrategy) HandleViolation(image string, pod *v1.Pod, violations []metadata.Vulnerability) error {
-	logrus.Warn("violations %v found in image %s, pod %s, ns %s", violations, image, pod.Name, pod.Namespace)
+func (l *LoggingStrategy) HandleViolation(image string, pod *v1.Pod, violations []securitypolicy.SecurityPolicyViolation) error {
+	if len(violations) == 0 {
+		return nil
+	}
+	logrus.Warnf("Found violations in pod %s, ns %s", pod.Name, pod.Namespace)
+	for _, v := range violations {
+		logrus.Warn(v.Reason)
+	}
 	return nil
 }
 
+// AnnotationStrategy adds "InvalidImageSecPolicy" as a label and an annotation, with detailed info in the annotations
+type AnnotationStrategy struct {
+}
+
+func (a *AnnotationStrategy) HandleViolation(image string, pod *v1.Pod, violations []securitypolicy.SecurityPolicyViolation) error {
+	// First, remove "InvalidImageSecPolicy" label/annotation in case it doesn't apply anymore
+	deletionKeys := []string{constants.InvalidImageSecPolicyLabel}
+	if err := pods.DeleteLabelsAndAnnotations(*pod, deletionKeys, deletionKeys); err != nil {
```

Comment:
does it return error when label and annotations does not exist?

Reply:
and you are trying to delete the annotations and label

Reply:
Yah, I delete any previous annotations/labels in case they no longer apply, and then add the new ones. If the annotation or label doesn't exist, it won't return an error.

Reply:
thanks.